Add an Allocation Attribute Edit Page

cecilialau6776 opened the pull request:
This PR adds a page that allows admins to edit allocation attributes and a button on the Allocation Detail page to access this page, which would close #672.

![image](https://github.com/user-attachments/assets/0718de2c-8239-46a4-98b5-74a059bb635d)
![image](https://github.com/user-attachments/assets/25fa06e2-856d-4687-9bbf-07e15efe38b9)

This PR also adds a signal for allocation attribute changes, which is sent when Allocation Change Requests are approved and through this page, which should [help] close #293.

cecilialau6776 commented:
@aebruno, I just wanted to ask if this was on the radar.

aebruno commented:
@cecilialau6776 Sorry for the delay. Can you rebase this PR to main then we can review. Looks good otherwise, Thanks!

cecilialau6776 commented:
Rebased and added tests!

cecilialau6776 commented:
@aebruno Yup! Squahsed and also rebased.

aebruno reviewed (changes requested):
@cecilialau6776 one minor suggestion. Can you remove the "Confirm" button if an allocation has no attributes to edit? For example:

<img width="1218" height="383" alt="2025-07-18_19-58" src="https://github.com/user-attachments/assets/8993a84a-1577-47c8-a7ec-01c38da38d78" />

If you click the confirm button, it errors out with a csfr error:

<img width="528" height="176" alt="2025-07-18_20-02" src="https://github.com/user-attachments/assets/79bdbea6-7b50-4142-bbb4-ee32066ede45" />

So maybe just have a cancel button to take the user back to previous page? 

Alternatively (or in addition) remove the "Edit Allocation Attributes" button from the allocations pages when no allocation attributes exist, for example here:

<img width="1236" height="292" alt="2025-07-18_20-01" src="https://github.com/user-attachments/assets/545a9aab-1f23-4bb1-add4-7b18b110f0d9" />

cecilialau6776 commented:
@aebruno Thanks, I didn't catch that when I was testing. I've removed the edit button and the confirm button when there are no allocation attributes.
Also noticed a typo for deleting allocation attributes and fixed it :)

cecilialau6776 commented:
Thanks for the feedback! I'd be happy to make the suggested changes. For several of them, I used examples from elsewhere in their respective files, but I can also update those if needed.

Eric-Butcher reviewed (commented):
Thanks for making these changes! It all looks great. Also I did not know that there was an HTTPStatus enum, I will have to use that myself!

I don't think it is necessary right now to go and change other code. 

At this stage we are looking to start paying down some of the technical debt in ColdFront, and part of that is making sure new code that comes in is a bit more maintianable than what we had previously. However, the idea was to not be rewriting stuff just yet until we have enough testing to be confidnt that those changes don't break anything.
